src/classes/COUR_CannotDelete_TEST.cls

```diff
@@ -0,0 +1,477 @@
+        }; 
+        insert terms; 
+        
+        Course_Offering__c courseOffering1 = UTIL_UnitTestData_TEST.createCourseOffering(courses[0].Id, terms[0].Id); 
+        
+        List<Plan_Requirement__c> planReqs = UTIL_UnitTestData_TEST.getMultipleTestPlanRequirements(1); 
+        planReqs[0].Course__c = courses[0].Id;
+        insert planReqs; 
+        
+        Test.startTest();
+        Database.DeleteResult[] results = Database.delete(courses, false);
+        Test.stopTest();
+        
+        List<Course__c> returnCourses = [SELECT Id
+                                        FROM Course__c
+                                        WHERE Id IN :courses]; 
+        System.assertEquals(0, returnCourses.size());
+    }
+    
+	/*********************************************************************************************************
+    * @description Tests the hasChildRecords method that the Course record has child records. 
+    */
+    @isTest
+    public static void testCourseHasChildRecords() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(new Hierarchy_Settings__c
+                                                            (Account_Processor__c = COUR_CannotDelete_TEST.adminAccRecTypeId,
+                                                            Prevent_Course_Deletion__c = True));
+        
+        //We're using "Administrative" record type for this new Account since
+        //we're not ready (we need to determine how best to handle retrieving recordTypeIds that are not
+        //in EDA Settings) to create a new method to retrieve the Universtiy_Department Account
+        //record type in UTIL_Describe. 
+        List<Account> accounts = UTIL_UnitTestData_TEST.getMultipleTestAccounts(1, COUR_CannotDelete_TEST.adminAccRecTypeId); 
+        insert accounts; 
+        
+        List<Course__c> courses = new List<Course__c>{
+            new Course__c (Name = 'Course1', Account__c  = accounts[0].Id),
+            new Course__c (Name = 'Course2', Account__c  = accounts[0].Id)
+        };        
+        insert courses; 
+        
+        List<Term__c> terms = new List<Term__c>{
+            new Term__c(Name = 'Fall', Account__c = accounts[0].Id),
+            new Term__c(Name = 'Spring', Account__c = accounts[0].Id)  
+        }; 
+        insert terms; 
+        
+        Course_Offering__c courseOffering1 = UTIL_UnitTestData_TEST.createCourseOffering(courses[0].Id, terms[0].Id); 
+        Course_Offering__c courseOffering2 = UTIL_UnitTestData_TEST.createCourseOffering(courses[1].Id, terms[1].Id); 
+        
+        List<Plan_Requirement__c> planReqs = UTIL_UnitTestData_TEST.getMultipleTestPlanRequirements(2); 
+        planReqs[0].Course__c = courses[0].Id;
+        planReqs[1].Course__c = courses[1].Id;
+        insert planReqs; 
+        
+        List<Course__c> returnCourses = [SELECT Id,
+                                        (SELECT ID, Name FROM Course_Offerings__r LIMIT 1),
```

Comment:
@kdy618 While this isn't a required change, you may want to make the use of Id or ID consistent within a single file.